### PR TITLE
Button chaser special (Vector Solution)

### PR DESCRIPTION
**Pull request by MarcRaeder**
Add Button-Chaser with Vector Solution

### CODE REVIEW

File: button-chaser-vector.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    fromCornerToMid();
+  }
+  else if (buttonRightSide >= (projectcontainer.offsetWidth - xBorder) && button.offsetTop >= (projectcontainer.offsetHeight - yBorder)) {
+    fromCornerToMid();
+  }
+}
+
+function fromCornerToMid() {
+
+  button.style.left = `${document.body.offsetWidth / 2}px`;
```
Comment (stefanluth) [inline button-chaser-vector.js]:
```suggestion
function fromCornerToMid() {
  button.style.left = `${document.body.offsetWidth / 2}px`;
```

File: button-chaser-vector.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+  else if (button.offsetLeft <= xBorder && button.offsetTop >= (projectcontainer.offsetHeight - yBorder)) {
+    fromCornerToMid();
+  }
+  else if (buttonRightSide >= (projectcontainer.offsetWidth - xBorder) && button.offsetTop <= yBorder) {
+    fromCornerToMid();
+  }
+  else if (buttonRightSide >= (projectcontainer.offsetWidth - xBorder) && button.offsetTop >= (projectcontainer.offsetHeight - yBorder)) {
+    fromCornerToMid();
+  }
```
Comment (stefanluth) [inline button-chaser-vector.js]: Move conditions to variables :)

File: button-chaser-vector.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const button = document.getElementsByClassName("button-to-chase")[0];
+const projectcontainer = document.getElementsByClassName("project-container")[0];
```
Comment (stefanluth) [inline button-chaser-vector.js]:
```suggestion
const projectContainer = document.getElementsByClassName("project-container")[0];
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: button-chaser-vector.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+const distanceBorder = 100;
+const xBorder = 10;
+const yBorder = 20;
+
+onmousemove = function (e) {
+
+  var buttonX = button.offsetLeft + (button.offsetWidth / 2);
+  var buttonY = button.offsetTop + (button.offsetHeight / 2);
+  var dX = e.clientX - buttonX;
```
Comment (DevBastiS) [inline button-chaser-vector.js]: what is d?

File: button-chaser-vector.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  var dY = e.clientY - buttonY;
+  var distance = Math.sqrt(Math.pow(dX, 2) + Math.pow(dY, 2));
```
Comment (DevBastiS) [inline button-chaser-vector.js]: distance from what?

File: button-chaser-vector.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  var buttonRightSide = button.offsetLeft + button.offsetWidth;
+
+  if (distance <= distanceBorder) {
+    button.style.left = `${button.offsetLeft - dX}px`;
+    button.style.top = `${button.offsetTop - dY}px`;
+  }
```
Comment (DevBastiS) [inline button-chaser-vector.js:23]:
```suggestion
  }
  
```

File: button-chaser-vector.js (hunk 2 above)
Comment (DevBastiS) [reply button-chaser-vector.js]: What @stefanluth says

**Review by DevBastiS — CHANGES_REQUESTED** [review]

**Review by stefanluth — APPROVED** [review]

File: button-chaser-vector.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  var buttonX = button.offsetLeft + (button.offsetWidth / 2);
+  var buttonY = button.offsetTop + (button.offsetHeight / 2);
+  var VectorXpositionMouseToButton = e.clientX - buttonX;
+  var VectorYpositionMouseToButton = e.clientY - buttonY;
+  var distanceMouseToButton = Math.sqrt(Math.pow(VectorXpositionMouseToButton, 2) + Math.pow(VectorYpositionMouseToButton, 2));
+  var buttonRightSide = button.offsetLeft + button.offsetWidth;
+  let buttonIsInTopLeftCorner = button.offsetLeft <= xBorder && button.offsetTop <= yBorder;
+  let buttonIsInBottomLeftCorner = button.offsetLeft <= xBorder && button.offsetTop >= (projectContainer.offsetHeight - yBorder);
+  let buttonIsInTopRightCorner = buttonRightSide >= (projectContainer.offsetWidth - xBorder) && button.offsetTop <= yBorder;
+  let buttonIsInBottomRightCorner = buttonRightSide >= (projectContainer.offsetWidth - xBorder) && button.offsetTop >= (projectContainer.offsetHeight - yBorder);
```
Comment (DevBastiS) [inline button-chaser-vector.js:18]:
```suggestion
  var buttonX = button.offsetLeft + (button.offsetWidth / 2);
  var buttonY = button.offsetTop + (button.offsetHeight / 2);
  
  var VectorXpositionMouseToButton = e.clientX - buttonX;
  var VectorYpositionMouseToButton = e.clientY - buttonY;
  
  var distanceMouseToButton = Math.sqrt(Math.pow(VectorXpositionMouseToButton, 2) + Math.pow(VectorYpositionMouseToButton, 2));
  
  var buttonRightSide = button.offsetLeft + button.offsetWidth;
  
  let buttonIsInTopLeftCorner = button.offsetLeft <= xBorder && button.offsetTop <= yBorder;
  let buttonIsInBottomLeftCorner = button.offsetLeft <= xBorder && button.offsetTop >= (projectContainer.offsetHeight - yBorder);
  let buttonIsInTopRightCorner = buttonRightSide >= (projectContainer.offsetWidth - xBorder) && button.offsetTop <= yBorder;
  let buttonIsInBottomRightCorner = buttonRightSide >= (projectContainer.offsetWidth - xBorder) && button.offsetTop >= (projectContainer.offsetHeight - yBorder);
```

**Review by DevBastiS — APPROVED** [review]

File: button-chaser-vector.js (hunk 7 above)
Comment (stefanluth) [inline button-chaser-vector.js:18]:
```
    var x = projectContainer.offsetHeight - yBorder;
```

Extract some of the calculations into variables (with proper names) to improve readability :)

**Comment by DevBastiS** [comment]
What Stefan said

**Pull request merged**